lxdprofile.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/set"
+	lxdapi "github.com/lxc/lxd/shared/api"
+	"gopkg.in/yaml.v2"
+)
+
+type LXDProfile struct {
+	lxdapi.Profile
```

Comment:
I'm still unsure about this, I wonder if we should copy the structs over so that we can adapt if required in the future. 

Note: the lxdapi.Profile is made of multiple composed structs rather than just one. It might be nice to have one, so it's easier to reason about.

Reply:
after conversation agreed on using equivalent of lxdapi.ProfilePut.